Refactor: Improve readability

JesusC-R opened the pull request:
Made small changes to the project:
1. Deleted the fragment in Footer.js since there's only one parent element.
2. Improved readability by renaming variables.

qu8n commented on src/utils/buildPieData.js:
```diff
@@ -12,7 +12,7 @@ export function buildPieData(rawData) {
     acc[`${duration} minutes`].sessions += 1;
     return acc;
-  }, durationShell);
+  }, initialDurationData);
```
I like this naming change. For consistency, let's also rename other instances of similar variable names (for example, renaming `attendanceShell` to `initialAttendanceData`) across the codebase.

JesusC-R replied:
On it

qu8n reviewed (changes requested):
@JesusC-R Thanks for opening this PR. Looking forward to your contributions 🙂

JesusC-R commented:
Made changes in the following files in my recent commit:

@/components/dashboard/Heading.
  1. Destrucure PropTypes to improve readability
  2. Replaced the div component with a fragment to avoid introducing an extra unnecessary DOM element.

@/components/dashboard/PieCharts
  1. Renamed the following variables:
    durationData, attendanceData, completionData

@/components/dashboard/RecentMilestones
  1. Encapsulated milestone calculation logic inside the calculateMilestones function.
  2. Refactored code to use the calculateMilestones function and MILESTONE_LEVELS_AND_UNITS object.
  3. Updated code to generate milestonesArr using filtered data and reverse operations.
  4. Improved code modularity, readability, and maintainability.

If there's a problem feel free to let me know.
